OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs

```diff
@@ -46,57 +46,34 @@ public class CreateEffectWarhead : Warhead
 		[Desc("Whether to consider actors in determining whether the explosion should happen. If false, only terrain will be considered.")]
 		public readonly bool ImpactActors = true;
 
-		static readonly BitSet<TargetableType> TargetTypeAir = new BitSet<TargetableType>("Air");
-
-		public ImpactType GetImpactType(World world, CPos cell, WPos pos, Actor firedBy)
-		{
-			// Matching target actor
-			if (ImpactActors)
-			{
-				var targetType = GetDirectHitTargetType(world, cell, pos, firedBy, true);
-				if (targetType == ImpactTargetType.ValidActor)
-					return ImpactType.TargetHit;
-				if (targetType == ImpactTargetType.InvalidActor)
-					return ImpactType.None;
-			}
+		[Desc("Can this warhead affect the actor that fired it. Defaults to 'true' for effect warheads.")]
+		public new readonly bool AffectsParent = true;
 
-			var dat = world.Map.DistanceAboveTerrain(pos);
-			if (dat > AirThreshold)
-				return ImpactType.Air;
-
-			return ImpactType.Ground;
-		}
+		static readonly BitSet<TargetableType> TargetTypeAir = new BitSet<TargetableType>("Air");
 
-		public ImpactTargetType GetDirectHitTargetType(World world, CPos cell, WPos pos, Actor firedBy, bool checkTargetValidity = false)
+		/// <summary>Checks if there are any actors at impact position and if the warhead is valid against any of them.</summary>
+		ImpactActorType ActorTypeAtImpact(World world, WPos pos, Actor firedBy)
 		{
-			var victims = world.FindActorsOnCircle(pos, WDist.Zero);
-			var invalidHit = false;
+			var anyInvalidActor = false;
 
-			foreach (var victim in victims)
+			// Check whether the impact position overlaps with an actor's hitshape
+			var potentialVictims = world.FindActorsOnCircle(pos, WDist.Zero);
```

Comment:
Style nit: this is relatively short and only used once, so could move inline on the `foreach`